ui/src/AddUserToGroup.js

```diff
@@ -0,0 +1,227 @@
+import React, { useEffect, useRef, useState } from "react";
+import { useHistory } from "react-router-dom";
+import { Helmet } from "react-helmet";
+import Announcer from "react-a11y-announcer";
+import { Link } from "react-router-dom";
+import Button from "./DesignSystemComponents/Button";
+import ErrorSummary from "./DesignSystemComponents/ErrorSummary";
+import Autosuggest from "react-autosuggest";
+import Parser from "html-react-parser";
+
+function AddUserToGroup(props) {
+  let history = useHistory();
+  let addUserInProgress = false;
+  const [errorSummary, setErrorSummary] = useState([]);
+  const errorSummaryTitle = useRef(null);
+  const [hasErrors, setHasErrors] = useState(false);
+  const [value, setValue] = useState("");
+  // Suggestions is tied to the AutoSuggest Component, it can be filted, mutated etc
+  const [suggestions, setSuggestions] = useState([]);
+  // We load userList this on the page load, we don't mutate it. We filter it to create Suggetions
+  const [userList, setUserList] = useState([]);
+
+  useEffect(() => {
+    async function fetchUsers() {
+      const response = await fetch(`/api/users?groupId=${props.groupId}`);
+      const usersJson = await response.json();
+
+      /*
+         React-AutoSuggest has the option of 'sections'
+         In order to look a little like the design system lets use this to 
+         create a Suggestions section
+      */
+      const users = [
+        {
+          title: "Suggestions",
+          users: usersJson,
+        },
+      ];
+
+      setUserList(users);
+      setSuggestions(users);
+    }
+
+    fetchUsers();
+  }, []);
+
+  function cancel() {
+    history.push(
+      `/groupadmin?groupId=${props.groupId}&groupName=${props.groupName}`
+    );
+  }
+
+  // The User can choose Add with a bad email address,
+  // Or by cutting and pasting a valid email address.
+  // Check it's in out list and get the UserId
+  function checkEmailExistsAndGetUserId(userInput) {
+    for (var i = 0; i < userList[0].users.length; i++) {
+      if (userList[0].users[i].email === userInput) {
+        return userList[0].users[i].id;
+      }
+    }
+
+    return undefined;
+  }
+
+  async function addUserToGroup() {
+    if (addUserInProgress) {
+      return;
+    }
+
+    addUserInProgress = true;
+
+    setErrorSummary("");
```

Comment:
Set to empty list instead of empty string?

Reply:
made it a string when declaring too